synnaxlabs/synnax
# Leo/sy 685 integration testing

# Feature Pull Request Template

## Key Information

- [SY-685](https://linear.app/synnaxlabs/issue/SY-685/integration-testing)

## Description

Integration testing framework, integration tests & GitHub actions CI workflow

## Revealed bugs

https://linear.app/synnaxlabs/issue/SY-1025/[cesium]-server-reading-incorrect-number-of-samples-revealed-by

## Basic Readiness Checklist

- [x] I have added relevant tests to cover the changes.
- [x] I have updated user facing documentation accordingly. - **None added**

## Manual QA Additions

- [ ] I have updated the [Release Candidate](/.github/PULL_REQUEST_TEMPLATE/rc.md) template
with necessary manual QA steps to test my change. - **Let's discuss whether the load tests should be part of QA requirements**

## Breaking Changes

Committed `pnpm.lock`, may need particular attention.

Also changed error message for when writers collide with each other and when opening a writer against an existing domain.
